src/features/network/components/network-select.tsx

```diff
@@ -3,37 +3,43 @@ import { Label } from '@/features/common/components/label'
 import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/features/common/components/select'
 import { cn } from '@/features/common/utils'
 import { useCallback } from 'react'
-import { useLocation, useNavigate } from 'react-router-dom'
-import { Urls } from '@/routes/urls'
+import { useLocation, useNavigate, useParams } from 'react-router-dom'
 
-const settingsUrl = Urls.Settings.build({})
+type NetworkSelectProps = {
+  showLabel?: boolean
+}
 
-export function NetworkSelect() {
+export function NetworkSelect({ showLabel = true }: NetworkSelectProps) {
   const [selectedNetwork, setSelectedNetwork] = useSelectedNetwork()
   const networkConfigs = useNetworkConfigs()
+  const { networkId: currentNetworkId } = useParams()
   const navigate = useNavigate()
   const location = useLocation()
 
   const handleNetworkChange = useCallback(
-    async (value: string) => {
-      if (location.pathname !== settingsUrl) {
-        navigate(settingsUrl)
+    (newNetworkId: string) => {
+      const currentPath = location.pathname
+      setSelectedNetwork(newNetworkId)
```

Comment:
```suggestion
      setSelectedNetwork(newNetworkId)
```
I don't believe this is needed. With this here, we end up calling setSelectedNetwork twice.

Reply:
I discovered why it's here. Without it you can select the network from the settings page or a page without network in the url. I wonder if there is a way to not call `setSelectedNetwork` twice?

Reply:
Perhaps I am not understanding. Where is the second time the `setSelectedNetwork` is being called? When you change the drop-down, the `setSelectedNetwork` is called once, and the URL is replaced with the new network, which ensures the URL that the selected network in state is the same.

Reply:
In the effect on the network-page. If you put a console.log just before both setSelectedNetwork calls, they both fire when changing the network.

Reply:
Oh, I see. Yeah, your're right, we don't need the `setSelectedNetwork` call in the NetworkSelect since the network-page will set the selected network because the URL network and state network will be different. I removed the `setSelectedNetwork` from the handleNetworkChange in the NetworkSelect component.

Reply:
@lempira I initially thought the same, however that only fires when your in a route that has the network in it. For example if you try change the network now from the settings page, it won't work.

Reply:
Ugh yeah. I just tried that. I reverted that change. I am not sure what's causing this either. I think the navigate is causing the components to unmount so on remounting in the new page, it's running the useEffect regardless to the deps. Perhaps the process is happening so quickly that selectedNetwork is not yet set in state so it's not returning early like it should. I don't see any perf hits from the double state setting. do you?

Reply:
I think that's exactly what's happening.
Yeah agreed, it'll have next to zero performance impact, so I think it's fine.